platforms/core-configuration/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/NamedDomainObjectContainerExtensions.kt

```diff
@@ -254,15 +255,23 @@ internal constructor(
      * @see [NamedDomainObjectProvider.configure]
      */
     operator fun String.invoke(configuration: T.() -> Unit): NamedDomainObjectProvider<T> =
-        this().apply { configure(configuration) }
+        named(this).apply { configure(configuration) }
 
     /**
      * Locates an object by name, without triggering its creation or configuration, failing if there is no such object.
      *
      * @see [NamedDomainObjectContainer.named]
      */
-    operator fun String.invoke(): NamedDomainObjectProvider<T> =
-        delegate.named(this)
+    @Deprecated("Use named(String) instead.", ReplaceWith("named(\"...\")"))
+    operator fun String.invoke(): NamedDomainObjectProvider<T> {
+        DeprecationLogger.deprecateBehaviour(String.format("Domain object '%s' found by String.invoke() notation.", this))
+            .withContext("The \"name\"() notation can cause confusion with methods provided by Kotlin or the JDK.")
+            .withAdvice("Use named(String) instead.")
```

Comment:
💭 Use backticks to indicate code?
```suggestion
            .withAdvice("Use `named(String)` instead.")
```

Reply:
I was tempted to do this too, but we don't do it in other places and I think that makes sense.

This message is going to appear in the console and build scans without any special Markdown formatting.  I think adding the backticks would be too much.